Automattic/jetpack
# ThreatsDataViews: Sort by status and severity by default

Fixes #
When viewing all threats, we default to sorting by severity. If we have an extensive history of critical severity threats, and current threats that are high or low severity, the current threats can only be found at the very end.

**Note:** It appears there is no way to apply custom sorting logic such as this to the baseView, only a single field can be sorted asc or desc.

## Proposed changes:

* Updating sorting of the `processedData` to sort by status first, then severity within the subsets.
* Show `Status` field by default when `item.status` is accessible (scan threats) so its clear the status of the threats we are viewing when viewing all.

### Other information:


## Jetpack product discussion


- https://github.com/Automattic/jetpack-scan-team/issues/1588

## Does this pull request change what data or activity we track or use?


- No

## Testing instructions:






* Checkout the `ThreatsDataViews` storybook, change the filters, ensure that threats are always sorted by status first, then severity
* Load this Protect branch using the Beta Tester, and add threats, ensure the sorting is correct when viewing current, historical and all threats
